cmd/kube-controller-manager/app/batch.go

```diff
@@ -46,6 +47,17 @@ func startCronJobController(ctx ControllerContext) (http.Handler, bool, error) {
 	if !ctx.AvailableResources[schema.GroupVersionResource{Group: "batch", Version: "v1beta1", Resource: "cronjobs"}] {
 		return nil, false, nil
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.CronJobControllerV2) {
+		cj2c, err := cronjob.NewControllerV2(ctx.InformerFactory.Batch().V1().Jobs(),
+			ctx.InformerFactory.Batch().V1beta1().CronJobs(),
+			ctx.ClientBuilder.ClientOrDie("cronjob-controller"),
+		)
+		if err != nil {
+			return nil, true, fmt.Errorf("error creating CronJob controller V2: %v", err)
+		}
+		go cj2c.Run(int(ctx.ComponentConfig.CronJobController.ConcurrentCronJobSyncs), ctx.Stop)
+		return nil, true, nil
+	}
 	cjc, err := cronjob.NewController(
```

Comment:
why do we start both controllers if the featuregate is off?  bug or intentional?  If intentional, you need a comment in the code explaining.

Reply:
> why do we start both controllers if the featuregate is off? bug or intentional? If intentional, you need a comment in the code explaining.

nm, I see the return